{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import Entrez\n",
    "Entrez.email = \"arturpedromartins@gmail.com\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1955 16539991\n"
     ]
    }
   ],
   "source": [
    "search_handle = Entrez.esearch(db=\"assembly\", term=\"(\\\"Salmonella\\\"[Organism] OR Salmonella[All Fields]) AND (latest[filter] AND \\\"complete genome\\\"[filter] AND all[filter] NOT anomalous[filter])\", usehistory=\"y\", idtype=\"acc\",retmax='100000')\n",
    "search_results = Entrez.read(search_handle)\n",
    "search_handle.close()\n",
    "\n",
    "idlist = search_results[\"IdList\"]\n",
    "\n",
    "print(len(idlist),idlist[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "handle gerado\n",
      "record gerado\n",
      "1955\n",
      "1955\n",
      "1955\n"
     ]
    }
   ],
   "source": [
    "import json\n",
    "idlist = search_results[\"IdList\"]\n",
    "handle = Entrez.esummary(db='assembly', id=idlist)\n",
    "print('handle gerado')\n",
    "record = Entrez.read(handle)\n",
    "print('record gerado')\n",
    "\n",
    "#json dos sumários de onde eu extraio os asm id e bio uid\n",
    "with open('all_record.json','w') as txt:\n",
    "    record_json = json.dumps(record,indent=1)\n",
    "\n",
    "    txt.write(record_json)\n",
    "summary = record['DocumentSummarySet']['DocumentSummary']\n",
    "asm_idlist = []\n",
    "bsm_idlist = []    \n",
    "for uid in summary:\n",
    "    asm_idlist.append(uid['AssemblyAccession'])\n",
    "    bsm_idlist.append(uid['BioSampleId'])\n",
    "print(len(asm_idlist))\n",
    "print(len(bsm_idlist))\n",
    "\n",
    "handle = Entrez.esummary(db='BioSample',id=bsm_idlist)\n",
    "record = Entrez.read(handle)\n",
    "# json com as informações que eu consigo do biosample\n",
    "with open('bsm_record.json','w') as txt:\n",
    "    record_json = json.dumps(record,indent=1)\n",
    "    txt.write(record_json)\n",
    "summary = record['DocumentSummarySet']['DocumentSummary']\n",
    "print(len(summary))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<BioSample access=\"public\" publication_date=\"2023-04-27T00:00:00.000\" last_update=\"2023-05-02T10:42:55.688\" submission_date=\"2023-04-27T13:56:07.213\" id=\"34404588\" accession=\"SAMN34404588\">\n",
      "  <Ids>\n",
      "    <Id db=\"BioSample\" is_primary=\"1\">SAMN34404588</Id>\n",
      "    <Id db_label=\"Sample name\">PNUSAS118467</Id>\n",
      "    <Id db=\"SRA\">SRS17448390</Id>\n",
      "  </Ids>\n",
      "  <Description>\n",
      "    <Title>Pathogen: clinical or host-associated sample from Salmonella enterica</Title>\n",
      "    <Organism taxonomy_id=\"28901\" taxonomy_name=\"Salmonella enterica\">\n",
      "      <OrganismName>Salmonella enterica</OrganismName>\n",
      "    </Organism>\n",
      "  </Description>\n",
      "  <Owner>\n",
      "    <Name abbreviation=\"UPHL_ID\">Utah Public Health Laboratory Infectious Disease submission group</Name>\n",
      "  </Owner>\n",
      "  <Models>\n",
      "    <Model>Pathogen.cl</Model>\n",
      "  </Models>\n",
      "  <Package display_name=\"Pathogen: clinical or host-associated; version 1.0\">Pathogen.cl.1.0</Package>\n",
      "  <Attributes>\n",
      "    <Attribute attribute_name=\"collected_by\" harmonized_name=\"collected_by\" display_name=\"collected by\">NA</Attribute>\n",
      "    <Attribute attribute_name=\"collection_date\" harmonized_name=\"collection_date\" display_name=\"collection date\">2019</Attribute>\n",
      "    <Attribute attribute_name=\"geo_loc_name\" harmonized_name=\"geo_loc_name\" display_name=\"geographic location\">USA</Attribute>\n",
      "    <Attribute attribute_name=\"host\" harmonized_name=\"host\" display_name=\"host\">Homo sapiens</Attribute>\n",
      "    <Attribute attribute_name=\"host_disease\" harmonized_name=\"host_disease\" display_name=\"host disease\">NA</Attribute>\n",
      "    <Attribute attribute_name=\"isolation_source\" harmonized_name=\"isolation_source\" display_name=\"isolation source\">Missing</Attribute>\n",
      "    <Attribute attribute_name=\"lat_lon\" harmonized_name=\"lat_lon\" display_name=\"latitude and longitude\">Missing</Attribute>\n",
      "    <Attribute attribute_name=\"strain\" harmonized_name=\"strain\" display_name=\"strain\">PNUSAS118467</Attribute>\n",
      "  </Attributes>\n",
      "  <Links>\n",
      "    <Link type=\"entrez\" target=\"bioproject\" label=\"PRJNA812595\">812595</Link>\n",
      "  </Links>\n",
      "  <Status status=\"live\" when=\"2023-04-27T13:56:07.214\" />\n",
      "</BioSample>\n"
     ]
    }
   ],
   "source": [
    "# xml com as informações que eu consigo no \"SampleData\" das informações do BioSample\n",
    "import xml.etree.ElementTree as ET\n",
    "\n",
    "element = ET.XML(summary[0][\"SampleData\"])\n",
    "ET.indent(element)\n",
    "print(ET.tostring(element, encoding='unicode'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['16528391', '16516701', '16498651', '16494151', '16494141', '16492711', '16472121', '16450151', '16436411', '16436401']\n",
      "SAMN07283941 7283941 \n",
      " GCA_010770995.3\n",
      "Title :  Pathogen: environmental/food/other sample from Salmonella enterica subsp. enterica serovar Bredeney\n",
      "Accession :  SAMN07283941\n",
      "Date :  2017/06/27\n",
      "PublicationDate :  2017/06/27\n",
      "ModificationDate :  2021/08/31\n",
      "Organization :  USDA FSIS\n",
      "Taxonomy :  134047\n",
      "Organism :  Salmonella enterica subsp. enterica serovar Bredeney\n",
      "SourceSample :  BioSample:SAMN07283941\n",
      "SampleData :  <BioSample access=\"public\" publication_date=\"2017-06-27T00:00:00.000\" last_update=\"2021-08-31T14:37:36.274\" submission_date=\"2017-06-27T15:19:56.850\" id=\"7283941\" accession=\"SAMN07283941\">   <Ids>     <Id db=\"BioSample\" is_primary=\"1\">SAMN07283941</Id>     <Id db_label=\"Sample name\">FSIS1702037</Id>     <Id db=\"SRA\">SRS2318664</Id>   </Ids>   <Description>     <Title>Pathogen: environmental/food/other sample from Salmonella enterica subsp. enterica serovar Bredeney</Title>     <Organism taxonomy_id=\"134047\" taxonomy_name=\"Salmonella enterica subsp. enterica serovar Bredeney\">       <OrganismName>Salmonella enterica subsp. enterica serovar Bredeney</OrganismName>     </Organism>   </Description>   <Owner>     <Name>USDA FSIS</Name>     <Contacts>       <Contact email=\"NCBISubmissions@fsis.usda.gov\">         <Name>           <First>Glenn</First>           <Last>Tillman</Last>         </Name>       </Contact>     </Contacts>   </Owner>   <Models>     <Model>Pathogen.env</Model>   </Models>   <Package display_name=\"Pathogen: environmental/food/other; version 1.0\">Pathogen.env.1.0</Package>   <Attributes>     <Attribute attribute_name=\"strain\" harmonized_name=\"strain\" display_name=\"strain\">FSIS1702037</Attribute>     <Attribute attribute_name=\"collected_by\" harmonized_name=\"collected_by\" display_name=\"collected by\">USDA-FSIS</Attribute>     <Attribute attribute_name=\"collection_date\" harmonized_name=\"collection_date\" display_name=\"collection date\">2017</Attribute>     <Attribute attribute_name=\"geo_loc_name\" harmonized_name=\"geo_loc_name\" display_name=\"geographic location\">USA:IL</Attribute>     <Attribute attribute_name=\"isolation_source\" harmonized_name=\"isolation_source\" display_name=\"isolation source\">animal-swine-market swine</Attribute>     <Attribute attribute_name=\"lat_lon\" harmonized_name=\"lat_lon\" display_name=\"latitude and longitude\">missing</Attribute>     <Attribute attribute_name=\"serovar\" harmonized_name=\"serovar\" display_name=\"serovar\">Bredeney</Attribute>     <Attribute attribute_name=\"sub_species\" harmonized_name=\"sub_species\" display_name=\"sub species\">enterica</Attribute>   </Attributes>   <Links>     <Link type=\"entrez\" target=\"bioproject\" label=\"PRJNA292666\">292666</Link>   </Links>   <Status status=\"live\" when=\"2017-06-27T15:19:56.850\"/> </BioSample> \n",
      "Identifiers :  BioSample: SAMN07283941; Sample name: FSIS1702037; SRA: SRS2318664\n",
      "Infraspecies :  strain: FSIS1702037\n",
      "Package :  Pathogen: environmental/food/other; version 1.0\n",
      "SortKey :  120210831\n",
      "<generator object DataHandler.parse at 0x7f34dd87c9e0>\n",
      "SAMN34310249 34310249 \n",
      " GCA_029906525.1\n",
      "Title :  Pathogen: environmental/food/other sample from Salmonella enterica subsp. enterica serovar Adelaide\n",
      "Accession :  SAMN34310249\n",
      "Date :  2023/04/22\n",
      "PublicationDate :  2023/04/22\n",
      "ModificationDate :  2023/04/24\n",
      "Organization :  BioAssets Corporation\n",
      "Taxonomy :  29473\n",
      "Organism :  Salmonella enterica subsp. enterica serovar Adelaide\n",
      "SourceSample :  BioSample:SAMN34310249\n",
      "SampleData :  <BioSample access=\"public\" publication_date=\"2023-04-22T00:00:00.000\" last_update=\"2023-04-24T13:48:13.126\" submission_date=\"2023-04-22T10:32:25.106\" id=\"34310249\" accession=\"SAMN34310249\">   <Ids>     <Id db=\"BioSample\" is_primary=\"1\">SAMN34310249</Id>     <Id db_label=\"Sample name\">Salmonella enterica subsp. enterica serovar Adelaide</Id>   </Ids>   <Description>     <Title>Pathogen: environmental/food/other sample from Salmonella enterica subsp. enterica serovar Adelaide</Title>     <Organism taxonomy_id=\"29473\" taxonomy_name=\"Salmonella enterica subsp. enterica serovar Adelaide\">       <OrganismName>Salmonella enterica subsp. enterica serovar Adelaide</OrganismName>     </Organism>   </Description>   <Owner>     <Name url=\"https://bioassets.com.ph\">BioAssets Corporation</Name>     <Contacts>       <Contact email=\"admontecillo@up.edu.ph\">         <Name>           <First>Andrew</First>           <Last>Montecillo</Last>           <Middle>Dungo</Middle>         </Name>       </Contact>     </Contacts>   </Owner>   <Models>     <Model>Pathogen.env</Model>   </Models>   <Package display_name=\"Pathogen: environmental/food/other; version 1.0\">Pathogen.env.1.0</Package>   <Attributes>     <Attribute attribute_name=\"collected_by\" harmonized_name=\"collected_by\" display_name=\"collected by\">Nelieza Tubeza, Laurence Vicera</Attribute>     <Attribute attribute_name=\"collection_date\" harmonized_name=\"collection_date\" display_name=\"collection date\">2023-01-23</Attribute>     <Attribute attribute_name=\"geo_loc_name\" harmonized_name=\"geo_loc_name\" display_name=\"geographic location\">Philippines: Los Banos, Laguna</Attribute>     <Attribute attribute_name=\"isolation_source\" harmonized_name=\"isolation_source\" display_name=\"isolation source\">eggshell rinse from eggs sold at select retail stores</Attribute>     <Attribute attribute_name=\"lat_lon\" harmonized_name=\"lat_lon\" display_name=\"latitude and longitude\">14.16 N 121.24 E</Attribute>     <Attribute attribute_name=\"serovar\" harmonized_name=\"serovar\" display_name=\"serovar\">Adelaide</Attribute>     <Attribute attribute_name=\"sub_species\" harmonized_name=\"sub_species\" display_name=\"sub species\">enterica</Attribute>     <Attribute attribute_name=\"Tax ID\">29473</Attribute>     <Attribute attribute_name=\"isolate\" harmonized_name=\"isolate\" display_name=\"isolate\">missing</Attribute>     <Attribute attribute_name=\"strain\" harmonized_name=\"strain\" display_name=\"strain\">H3-X1</Attribute>   </Attributes>   <Links>     <Link type=\"entrez\" target=\"bioproject\" label=\"PRJNA959347\">959347</Link>   </Links>   <Status status=\"live\" when=\"2023-04-22T10:32:25.109\"/> </BioSample> \n",
      "Identifiers :  BioSample: SAMN34310249; Sample name: Salmonella enterica subsp. enterica serovar Adelaide\n",
      "Infraspecies :  strain: H3-X1\n",
      "Package :  Pathogen: environmental/food/other; version 1.0\n",
      "SortKey :  120230424\n",
      "<generator object DataHandler.parse at 0x7f34dd87c2e0>\n",
      "SAMN26754305 26754305 \n",
      " GCF_029866625.1\n",
      "Title :  Pathogen: environmental/food/other sample from Salmonella enterica sub species enterica serovar Enteritidis\n",
      "Accession :  SAMN26754305\n",
      "Date :  2023/03/19\n",
      "PublicationDate :  2023/03/19\n",
      "ModificationDate :  2023/03/19\n",
      "Organization :  Ludong University\n",
      "Taxonomy :  149539\n",
      "Organism :  Salmonella enterica subsp. enterica serovar Enteritidis\n",
      "SourceSample :  BioSample:SAMN26754305\n",
      "SampleData :  <BioSample access=\"public\" publication_date=\"2023-03-19T00:50:12.737\" last_update=\"2023-03-19T00:50:12.737\" submission_date=\"2022-03-17T20:41:03.540\" id=\"26754305\" accession=\"SAMN26754305\">   <Ids>     <Id db=\"BioSample\" is_primary=\"1\">SAMN26754305</Id>     <Id db_label=\"Sample name\">Jilin228</Id>   </Ids>   <Description>     <Title>Pathogen: environmental/food/other sample from Salmonella enterica sub species enterica serovar Enteritidis</Title>     <Organism taxonomy_id=\"149539\" taxonomy_name=\"Salmonella enterica subsp. enterica serovar Enteritidis\">       <OrganismName>Salmonella enterica subsp. enterica serovar Enteritidis</OrganismName>     </Organism>   </Description>   <Owner>     <Name>Ludong University</Name>     <Contacts>       <Contact email=\"hwzhu@ldu.edu.cn\">         <Name>           <First>Hongwei</First>           <Last>Zhu</Last>         </Name>       </Contact>     </Contacts>   </Owner>   <Models>     <Model>Pathogen.env</Model>   </Models>   <Package display_name=\"Pathogen: environmental/food/other; version 1.0\">Pathogen.env.1.0</Package>   <Attributes>     <Attribute attribute_name=\"strain\" harmonized_name=\"strain\" display_name=\"strain\">JL228</Attribute>     <Attribute attribute_name=\"collected_by\" harmonized_name=\"collected_by\" display_name=\"collected by\">Xin Yu</Attribute>     <Attribute attribute_name=\"collection_date\" harmonized_name=\"collection_date\" display_name=\"collection date\">2018-03-15</Attribute>     <Attribute attribute_name=\"geo_loc_name\" harmonized_name=\"geo_loc_name\" display_name=\"geographic location\">China</Attribute>     <Attribute attribute_name=\"isolation_source\" harmonized_name=\"isolation_source\" display_name=\"isolation source\">rabbit liver</Attribute>     <Attribute attribute_name=\"lat_lon\" harmonized_name=\"lat_lon\" display_name=\"latitude and longitude\">44.424137 N 125.191539 E</Attribute>     <Attribute attribute_name=\"culture_collection\" harmonized_name=\"culture_collection\" display_name=\"culture collection\">not applicable</Attribute>     <Attribute attribute_name=\"genotype\" harmonized_name=\"genotype\" display_name=\"genotype\">I</Attribute>     <Attribute attribute_name=\"serovar\" harmonized_name=\"serovar\" display_name=\"serovar\">Enteritidis</Attribute>     <Attribute attribute_name=\"sub_species\" harmonized_name=\"sub_species\" display_name=\"sub species\">enterica</Attribute>   </Attributes>   <Links>     <Link type=\"entrez\" target=\"bioproject\" label=\"PRJNA817289\">817289</Link>   </Links>   <Status status=\"live\" when=\"2023-03-19T00:50:12.737\"/> </BioSample> \n",
      "Identifiers :  BioSample: SAMN26754305; Sample name: Jilin228\n",
      "Infraspecies :  strain: JL228\n",
      "Package :  Pathogen: environmental/food/other; version 1.0\n",
      "SortKey :  120230319\n",
      "<generator object DataHandler.parse at 0x7f34dd44af90>\n",
      "SAMN16532586 16532586 \n",
      " GCA_014946605.2\n",
      "Title :  Pathogen: environmental/food/other sample from Salmonella enterica subsp. enterica serovar Newport\n",
      "Accession :  SAMN16532586\n",
      "Date :  2020/10/23\n",
      "PublicationDate :  2020/10/23\n",
      "ModificationDate :  2020/10/23\n",
      "Organization :  United States Department of Agriculture, Food Safety and Inspection Service\n",
      "Taxonomy :  108619\n",
      "Organism :  Salmonella enterica subsp. enterica serovar Newport\n",
      "SourceSample :  BioSample:SAMN16532586\n",
      "SampleData :  <BioSample access=\"public\" publication_date=\"2020-10-23T00:00:00.000\" last_update=\"2020-10-23T13:48:18.753\" submission_date=\"2020-10-23T13:46:06.073\" id=\"16532586\" accession=\"SAMN16532586\">   <Ids>     <Id db=\"BioSample\" is_primary=\"1\">SAMN16532586</Id>     <Id db=\"Sample name\">FSIS12034723</Id>     <Id db=\"SRA\">SRS7573762</Id>   </Ids>   <Description>     <Title>Pathogen: environmental/food/other sample from Salmonella enterica subsp. enterica serovar Newport</Title>     <Organism taxonomy_id=\"108619\" taxonomy_name=\"Salmonella enterica subsp. enterica serovar Newport\">       <OrganismName>Salmonella enterica subsp. enterica serovar Newport</OrganismName>     </Organism>   </Description>   <Owner>     <Name abbreviation=\"USDA-FSIS\" url=\"https://www.fsis.usda.gov\">United States Department of Agriculture, Food Safety and Inspection Service</Name>   </Owner>   <Models>     <Model>Pathogen.env</Model>   </Models>   <Package display_name=\"Pathogen: environmental/food/other; version 1.0\">Pathogen.env.1.0</Package>   <Attributes>     <Attribute attribute_name=\"strain\" harmonized_name=\"strain\" display_name=\"strain\">FSIS12034723</Attribute>     <Attribute attribute_name=\"collected_by\" harmonized_name=\"collected_by\" display_name=\"collected by\">USDA-FSIS</Attribute>     <Attribute attribute_name=\"collection_date\" harmonized_name=\"collection_date\" display_name=\"collection date\">2020</Attribute>     <Attribute attribute_name=\"isolation_source\" harmonized_name=\"isolation_source\" display_name=\"isolation source\">comminuted beef</Attribute>     <Attribute attribute_name=\"geo_loc_name\" harmonized_name=\"geo_loc_name\" display_name=\"geographic location\">USA:TX</Attribute>     <Attribute attribute_name=\"lat_lon\" harmonized_name=\"lat_lon\" display_name=\"latitude and longitude\">missing</Attribute>     <Attribute attribute_name=\"serovar\" harmonized_name=\"serovar\" display_name=\"serovar\">Salmonella enterica subsp. enterica serovar Newport</Attribute>     <Attribute attribute_name=\"sub_species\" harmonized_name=\"sub_species\" display_name=\"sub species\">enterica</Attribute>   </Attributes>   <Links>     <Link type=\"entrez\" target=\"bioproject\" label=\"PRJNA242847\">242847</Link>   </Links>   <Status status=\"live\" when=\"2020-10-23T13:46:06.073\"/> </BioSample> \n",
      "Identifiers :  BioSample: SAMN16532586; Sample name: FSIS12034723; SRA: SRS7573762\n",
      "Infraspecies :  strain: FSIS12034723\n",
      "Package :  Pathogen: environmental/food/other; version 1.0\n",
      "SortKey :  120201023\n",
      "<generator object DataHandler.parse at 0x7f34dd791350>\n",
      "SAMN10924178 10924178 \n",
      " GCA_007425265.2\n",
      "Title :  Pathogen: environmental/food/other sample from Salmonella enterica subsp. enterica serovar 4,[5],12:i:-\n",
      "Accession :  SAMN10924178\n",
      "Date :  2019/02/13\n",
      "PublicationDate :  2019/02/13\n",
      "ModificationDate :  2019/06/06\n",
      "Organization :  USDA FSIS\n",
      "Taxonomy :  440524\n",
      "Organism :  Salmonella enterica subsp. enterica serovar 4,[5],12:i:-\n",
      "SourceSample :  BioSample:SAMN10924178\n",
      "SampleData :  <BioSample access=\"public\" publication_date=\"2019-02-13T00:00:00.000\" last_update=\"2019-06-06T12:22:01.328\" submission_date=\"2019-02-13T07:53:04.410\" id=\"10924178\" accession=\"SAMN10924178\">   <Ids>     <Id db=\"BioSample\" is_primary=\"1\">SAMN10924178</Id>     <Id db_label=\"Sample name\">FSIS31901558</Id>     <Id db=\"SRA\">SRS4361102</Id>   </Ids>   <Description>     <Title>Pathogen: environmental/food/other sample from Salmonella enterica subsp. enterica serovar 4,[5],12:i:-</Title>     <Organism taxonomy_id=\"440524\" taxonomy_name=\"Salmonella enterica subsp. enterica serovar 4,[5],12:i:-\">       <OrganismName>Salmonella enterica subsp. enterica serovar 4,[5],12:i:-</OrganismName>     </Organism>   </Description>   <Owner>     <Name>USDA FSIS</Name>     <Contacts>       <Contact email=\"NCBISubmissions@fsis.usda.gov\">         <Name>           <First>Glenn</First>           <Last>Tillman</Last>         </Name>       </Contact>     </Contacts>   </Owner>   <Models>     <Model>Pathogen.env</Model>   </Models>   <Package display_name=\"Pathogen: environmental/food/other; version 1.0\">Pathogen.env.1.0</Package>   <Attributes>     <Attribute attribute_name=\"strain\" harmonized_name=\"strain\" display_name=\"strain\">FSIS31901558</Attribute>     <Attribute attribute_name=\"collected_by\" harmonized_name=\"collected_by\" display_name=\"collected by\">USDA-FSIS</Attribute>     <Attribute attribute_name=\"collection_date\" harmonized_name=\"collection_date\" display_name=\"collection date\">2019</Attribute>     <Attribute attribute_name=\"geo_loc_name\" harmonized_name=\"geo_loc_name\" display_name=\"geographic location\">USA:NY</Attribute>     <Attribute attribute_name=\"lat_lon\" harmonized_name=\"lat_lon\" display_name=\"latitude and longitude\">missing</Attribute>     <Attribute attribute_name=\"isolation_source\" harmonized_name=\"isolation_source\" display_name=\"isolation source\">Product-Raw-Ground, Comminuted or Otherwise Nonintact-Pork</Attribute>     <Attribute attribute_name=\"serovar\" harmonized_name=\"serovar\" display_name=\"serovar\">4,[5],12:i:-</Attribute>     <Attribute attribute_name=\"sub_species\" harmonized_name=\"sub_species\" display_name=\"sub species\">enterica</Attribute>   </Attributes>   <Links>     <Link type=\"entrez\" target=\"bioproject\" label=\"PRJNA242847\">242847</Link>   </Links>   <Status status=\"live\" when=\"2019-02-13T07:53:04.412\"/> </BioSample> \n",
      "Identifiers :  BioSample: SAMN10924178; Sample name: FSIS31901558; SRA: SRS4361102\n",
      "Infraspecies :  strain: FSIS31901558\n",
      "Package :  Pathogen: environmental/food/other; version 1.0\n",
      "SortKey :  120190606\n",
      "<generator object DataHandler.parse at 0x7f34dd63f430>\n",
      "SAMN33848916 33848916 \n",
      " GCA_029854355.1\n",
      "Title :  Pathogen: clinical or host-associated sample from Salmonella enterica subsp. diarizonae serovar 48:i:z\n",
      "Accession :  SAMN33848916\n",
      "Date :  2023/03/22\n",
      "PublicationDate :  2023/03/22\n",
      "ModificationDate :  2023/04/14\n",
      "Organization :  NORTH-WEST UNIVERSITY\n",
      "Taxonomy :  1192842\n",
      "Organism :  Salmonella enterica subsp. diarizonae serovar 48:i:z\n",
      "SourceSample :  BioSample:SAMN33848916\n",
      "SampleData :  <BioSample access=\"public\" publication_date=\"2023-03-22T00:00:00.000\" last_update=\"2023-04-14T08:34:41.803\" submission_date=\"2023-03-22T03:10:05.240\" id=\"33848916\" accession=\"SAMN33848916\">   <Ids>     <Id db=\"BioSample\" is_primary=\"1\">SAMN33848916</Id>     <Id db_label=\"Sample name\">NWU_5</Id>     <Id db=\"SRA\">SRS17110695</Id>   </Ids>   <Description>     <Title>Pathogen: clinical or host-associated sample from Salmonella enterica subsp. diarizonae serovar 48:i:z</Title>     <Organism taxonomy_id=\"1192842\" taxonomy_name=\"Salmonella enterica subsp. diarizonae serovar 48:i:z\">       <OrganismName>Salmonella enterica subsp. diarizonae serovar 48:i:z</OrganismName>     </Organism>   </Description>   <Owner>     <Name url=\"https://nwu.ac.za\">NORTH-WEST UNIVERSITY</Name>     <Contacts>       <Contact email=\"tesleemabolarinwa@gmail.com\">         <Name>           <First>Tesleem</First>           <Last>Abolarinwa</Last>           <Middle>Olatunde</Middle>         </Name>       </Contact>     </Contacts>   </Owner>   <Models>     <Model>Pathogen.cl</Model>   </Models>   <Package display_name=\"Pathogen: clinical or host-associated; version 1.0\">Pathogen.cl.1.0</Package>   <Attributes>     <Attribute attribute_name=\"strain\" harmonized_name=\"strain\" display_name=\"strain\">NWU_5</Attribute>     <Attribute attribute_name=\"collected_by\" harmonized_name=\"collected_by\" display_name=\"collected by\">Abolarinwa T.O.</Attribute>     <Attribute attribute_name=\"collection_date\" harmonized_name=\"collection_date\" display_name=\"collection date\">2021-04-20</Attribute>     <Attribute attribute_name=\"host\" harmonized_name=\"host\" display_name=\"host\">Bos taurus</Attribute>     <Attribute attribute_name=\"host_disease\" harmonized_name=\"host_disease\" display_name=\"host disease\">Salmonella gastroenteritis</Attribute>     <Attribute attribute_name=\"lat_lon\" harmonized_name=\"lat_lon\" display_name=\"latitude and longitude\">27 S 26 E</Attribute>     <Attribute attribute_name=\"Tax ID\">1192842</Attribute>     <Attribute attribute_name=\"serovar\" harmonized_name=\"serovar\" display_name=\"serovar\">48:i:z</Attribute>     <Attribute attribute_name=\"sub_species\" harmonized_name=\"sub_species\" display_name=\"sub species\">diarizonae</Attribute>     <Attribute attribute_name=\"country\" harmonized_name=\"geo_loc_name\" display_name=\"geographic location\">South Africa: North West University farm</Attribute>     <Attribute attribute_name=\"isolation_source\" harmonized_name=\"isolation_source\" display_name=\"isolation source\">missing</Attribute>   </Attributes>   <Links>     <Link type=\"entrez\" target=\"bioproject\" label=\"PRJNA947358\">947358</Link>   </Links>   <Status status=\"live\" when=\"2023-03-22T03:10:05.243\"/> </BioSample> \n",
      "Identifiers :  BioSample: SAMN33848916; Sample name: NWU_5; SRA: SRS17110695\n",
      "Infraspecies :  strain: NWU_5\n",
      "Package :  Pathogen: clinical or host-associated; version 1.0\n",
      "SortKey :  120230414\n",
      "<generator object DataHandler.parse at 0x7f34dd438120>\n",
      "SAMN34003877 34003877 \n",
      " GCF_029814235.1\n",
      "Title :  Microbe sample from Salmonella enterica\n",
      "Accession :  SAMN34003877\n",
      "Date :  2023/03/31\n",
      "PublicationDate :  2023/03/31\n",
      "ModificationDate :  2023/04/03\n",
      "Organization :  Fenyang college of shanxi medical university\n",
      "Taxonomy :  28901\n",
      "Organism :  Salmonella enterica\n",
      "SourceSample :  BioSample:SAMN34003877\n",
      "SampleData :  <BioSample access=\"public\" publication_date=\"2023-03-31T00:00:00.000\" last_update=\"2023-04-03T12:45:19.569\" submission_date=\"2023-03-31T04:33:04.043\" id=\"34003877\" accession=\"SAMN34003877\">   <Ids>     <Id db=\"BioSample\" is_primary=\"1\">SAMN34003877</Id>     <Id db_label=\"Sample name\">27A</Id>   </Ids>   <Description>     <Title>Microbe sample from Salmonella enterica</Title>     <Organism taxonomy_id=\"28901\" taxonomy_name=\"Salmonella enterica\">       <OrganismName>Salmonella enterica</OrganismName>     </Organism>   </Description>   <Owner>     <Name>Fenyang college of shanxi medical university</Name>     <Contacts>       <Contact email=\"bj0726@sxmu.edu.cn\">         <Name>           <First>benjin</First>           <Last>xu</Last>         </Name>       </Contact>     </Contacts>   </Owner>   <Models>     <Model>Microbe, viral or environmental</Model>   </Models>   <Package display_name=\"Microbe; version 1.0\">Microbe.1.0</Package>   <Attributes>     <Attribute attribute_name=\"strain\" harmonized_name=\"strain\" display_name=\"strain\">27A</Attribute>     <Attribute attribute_name=\"host\" harmonized_name=\"host\" display_name=\"host\">Homo sapiens</Attribute>     <Attribute attribute_name=\"isolation_source\" harmonized_name=\"isolation_source\" display_name=\"isolation source\">feces</Attribute>     <Attribute attribute_name=\"collection_date\" harmonized_name=\"collection_date\" display_name=\"collection date\">2021-05-30</Attribute>     <Attribute attribute_name=\"geo_loc_name\" harmonized_name=\"geo_loc_name\" display_name=\"geographic location\">China: Fenyang, Shanxi</Attribute>     <Attribute attribute_name=\"sample_type\" harmonized_name=\"sample_type\" display_name=\"sample type\">tissue sample</Attribute>   </Attributes>   <Links>     <Link type=\"entrez\" target=\"bioproject\" label=\"PRJNA950588\">950588</Link>   </Links>   <Status status=\"live\" when=\"2023-03-31T04:33:04.045\"/> </BioSample> \n",
      "Identifiers :  BioSample: SAMN34003877; Sample name: 27A\n",
      "Infraspecies :  strain: 27A\n",
      "Package :  Microbe; version 1.0\n",
      "SortKey :  120230403\n",
      "<generator object DataHandler.parse at 0x7f34dd438120>\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m/tmp/ipykernel_18252/303489594.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     16\u001b[0m \u001b[0;31m# Fetch the records and print them\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     17\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mrecord_id\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mid_list\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 18\u001b[0;31m     \u001b[0mhandle\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mEntrez\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mesummary\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdb\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdb\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mid\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mrecord_id\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     19\u001b[0m     \u001b[0mrecord\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mEntrez\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mhandle\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     20\u001b[0m     \u001b[0msummary\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mrecord\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'DocumentSummarySet'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'DocumentSummary'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.9/site-packages/Bio/Entrez/__init__.py\u001b[0m in \u001b[0;36mesummary\u001b[0;34m(**keywds)\u001b[0m\n\u001b[1;32m    334\u001b[0m     \u001b[0mvariables\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mupdate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkeywds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    335\u001b[0m     \u001b[0mrequest\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_build_request\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcgi\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvariables\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 336\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0m_open\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrequest\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    337\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    338\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.9/site-packages/Bio/Entrez/__init__.py\u001b[0m in \u001b[0;36m_open\u001b[0;34m(request)\u001b[0m\n\u001b[1;32m    592\u001b[0m     \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mrange\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmax_tries\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    593\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 594\u001b[0;31m             \u001b[0mhandle\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0murlopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrequest\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    595\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mHTTPError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mexception\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    596\u001b[0m             \u001b[0;31m# Reraise if the final try fails\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.9/urllib/request.py\u001b[0m in \u001b[0;36murlopen\u001b[0;34m(url, data, timeout, cafile, capath, cadefault, context)\u001b[0m\n\u001b[1;32m    212\u001b[0m     \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    213\u001b[0m         \u001b[0mopener\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_opener\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 214\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0mopener\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtimeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    215\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    216\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0minstall_opener\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mopener\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.9/urllib/request.py\u001b[0m in \u001b[0;36mopen\u001b[0;34m(self, fullurl, data, timeout)\u001b[0m\n\u001b[1;32m    515\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    516\u001b[0m         \u001b[0msys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0maudit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'urllib.Request'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mreq\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfull_url\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mreq\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mreq\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mheaders\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mreq\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_method\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 517\u001b[0;31m         \u001b[0mresponse\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_open\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mreq\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    518\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    519\u001b[0m         \u001b[0;31m# post-process response\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.9/urllib/request.py\u001b[0m in \u001b[0;36m_open\u001b[0;34m(self, req, data)\u001b[0m\n\u001b[1;32m    532\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    533\u001b[0m         \u001b[0mprotocol\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mreq\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtype\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 534\u001b[0;31m         result = self._call_chain(self.handle_open, protocol, protocol +\n\u001b[0m\u001b[1;32m    535\u001b[0m                                   '_open', req)\n\u001b[1;32m    536\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.9/urllib/request.py\u001b[0m in \u001b[0;36m_call_chain\u001b[0;34m(self, chain, kind, meth_name, *args)\u001b[0m\n\u001b[1;32m    492\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0mhandler\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mhandlers\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    493\u001b[0m             \u001b[0mfunc\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mgetattr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mhandler\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmeth_name\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 494\u001b[0;31m             \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfunc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    495\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mresult\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    496\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.9/urllib/request.py\u001b[0m in \u001b[0;36mhttps_open\u001b[0;34m(self, req)\u001b[0m\n\u001b[1;32m   1387\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1388\u001b[0m         \u001b[0;32mdef\u001b[0m \u001b[0mhttps_open\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mreq\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1389\u001b[0;31m             return self.do_open(http.client.HTTPSConnection, req,\n\u001b[0m\u001b[1;32m   1390\u001b[0m                 context=self._context, check_hostname=self._check_hostname)\n\u001b[1;32m   1391\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.9/urllib/request.py\u001b[0m in \u001b[0;36mdo_open\u001b[0;34m(self, http_class, req, **http_conn_args)\u001b[0m\n\u001b[1;32m   1344\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1345\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1346\u001b[0;31m                 h.request(req.get_method(), req.selector, req.data, headers,\n\u001b[0m\u001b[1;32m   1347\u001b[0m                           encode_chunked=req.has_header('Transfer-encoding'))\n\u001b[1;32m   1348\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mOSError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0merr\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;31m# timeout error\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.9/http/client.py\u001b[0m in \u001b[0;36mrequest\u001b[0;34m(self, method, url, body, headers, encode_chunked)\u001b[0m\n\u001b[1;32m   1283\u001b[0m                 encode_chunked=False):\n\u001b[1;32m   1284\u001b[0m         \u001b[0;34m\"\"\"Send a complete request to the server.\"\"\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1285\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_send_request\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmethod\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbody\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mheaders\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mencode_chunked\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1286\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1287\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_send_request\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmethod\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbody\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mheaders\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mencode_chunked\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.9/http/client.py\u001b[0m in \u001b[0;36m_send_request\u001b[0;34m(self, method, url, body, headers, encode_chunked)\u001b[0m\n\u001b[1;32m   1329\u001b[0m             \u001b[0;31m# default charset of iso-8859-1.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1330\u001b[0m             \u001b[0mbody\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_encode\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mbody\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'body'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1331\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mendheaders\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mbody\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mencode_chunked\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mencode_chunked\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1332\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1333\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mgetresponse\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.9/http/client.py\u001b[0m in \u001b[0;36mendheaders\u001b[0;34m(self, message_body, encode_chunked)\u001b[0m\n\u001b[1;32m   1278\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1279\u001b[0m             \u001b[0;32mraise\u001b[0m \u001b[0mCannotSendHeader\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1280\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_send_output\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmessage_body\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mencode_chunked\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mencode_chunked\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1281\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1282\u001b[0m     def request(self, method, url, body=None, headers={}, *,\n",
      "\u001b[0;32m~/anaconda3/lib/python3.9/http/client.py\u001b[0m in \u001b[0;36m_send_output\u001b[0;34m(self, message_body, encode_chunked)\u001b[0m\n\u001b[1;32m   1038\u001b[0m         \u001b[0mmsg\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34mb\"\\r\\n\"\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_buffer\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1039\u001b[0m         \u001b[0;32mdel\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_buffer\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1040\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmsg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1041\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1042\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mmessage_body\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.9/http/client.py\u001b[0m in \u001b[0;36msend\u001b[0;34m(self, data)\u001b[0m\n\u001b[1;32m    978\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msock\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    979\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mauto_open\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 980\u001b[0;31m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mconnect\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    981\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    982\u001b[0m                 \u001b[0;32mraise\u001b[0m \u001b[0mNotConnected\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.9/http/client.py\u001b[0m in \u001b[0;36mconnect\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   1452\u001b[0m                 \u001b[0mserver_hostname\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhost\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1453\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1454\u001b[0;31m             self.sock = self._context.wrap_socket(self.sock,\n\u001b[0m\u001b[1;32m   1455\u001b[0m                                                   server_hostname=server_hostname)\n\u001b[1;32m   1456\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.9/ssl.py\u001b[0m in \u001b[0;36mwrap_socket\u001b[0;34m(self, sock, server_side, do_handshake_on_connect, suppress_ragged_eofs, server_hostname, session)\u001b[0m\n\u001b[1;32m    499\u001b[0m         \u001b[0;31m# SSLSocket class handles server_hostname encoding before it calls\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    500\u001b[0m         \u001b[0;31m# ctx._wrap_socket()\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 501\u001b[0;31m         return self.sslsocket_class._create(\n\u001b[0m\u001b[1;32m    502\u001b[0m             \u001b[0msock\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0msock\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    503\u001b[0m             \u001b[0mserver_side\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mserver_side\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.9/ssl.py\u001b[0m in \u001b[0;36m_create\u001b[0;34m(cls, sock, server_side, do_handshake_on_connect, suppress_ragged_eofs, server_hostname, context, session)\u001b[0m\n\u001b[1;32m   1039\u001b[0m                         \u001b[0;31m# non-blocking\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1040\u001b[0m                         \u001b[0;32mraise\u001b[0m \u001b[0mValueError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"do_handshake_on_connect should not be specified for non-blocking sockets\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1041\u001b[0;31m                     \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdo_handshake\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1042\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mOSError\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mValueError\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1043\u001b[0m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mclose\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.9/ssl.py\u001b[0m in \u001b[0;36mdo_handshake\u001b[0;34m(self, block)\u001b[0m\n\u001b[1;32m   1308\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mtimeout\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;36m0.0\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mblock\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1309\u001b[0m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msettimeout\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1310\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_sslobj\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdo_handshake\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1311\u001b[0m         \u001b[0;32mfinally\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1312\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msettimeout\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtimeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "# Set the email address to use for the Entrez API\n",
    "Entrez.email = \"your.email@example.com\"\n",
    "\n",
    "# Set the search parameters\n",
    "search_term = \"Salmonella_Enterica[Orgn] and \\\"complete genome\\\"[filter]\"\n",
    "db = \"assembly\"\n",
    "retmax = 10  # max number of records to return\n",
    "\n",
    "# Search for records using the Entrez API\n",
    "handle = Entrez.esearch(db=db, term=search_term, retmax=retmax)\n",
    "record = Entrez.read(handle)\n",
    "\n",
    "# Get the list of record IDs from the search results\n",
    "id_list = record[\"IdList\"]\n",
    "print(id_list)\n",
    "# Fetch the records and print them\n",
    "for record_id in id_list:\n",
    "    handle = Entrez.esummary(db=db, id=record_id)\n",
    "    record = Entrez.read(handle)\n",
    "    summary = record['DocumentSummarySet']['DocumentSummary'][0]\n",
    "    print(summary['BioSampleAccn'],summary['BioSampleId'],\"\\n\",summary['AssemblyAccession'])\n",
    "    handle = Entrez.esummary(db='BioSample', id=summary['BioSampleId'])\n",
    "    record = Entrez.read(handle)\n",
    "    summary = record['DocumentSummarySet']['DocumentSummary'][0]\n",
    "    for i in summary.keys():\n",
    "        print(i,\": \", summary[i])\n",
    "    \n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.2"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
